packages/template-block/src/hooks/useTemplateBlockData.ts

```diff
@@ -0,0 +1,147 @@
+        textRatio,
+        textAnchoringHorizontal,
+        textAnchoringVertical,
+    } = blockSettings;
+    const { previewCustom } = blockAssets;
+
+    const hasPreview = preview !== PreviewType.None;
+    const flexDirectionStyles = hasPreview
+        ? textPositioningToStyles[textPositioning]
+        : textPositioningToStyles[TextPositioningType.Right];
+    const isRows = getIsRows(hasPreview, textPositioning);
+    const borderRadius = hasRadius_blockCard ? radiusValue_blockCard : radiusStyleMap[radiusChoice_blockCard];
+    const border = hasBorder_blockCard
+        ? `${borderWidth_blockCard} ${borderStyle_blockCard} ${toRgbaString(borderColor_blockCard)}`
+        : 'none';
+
+    const saveDescription = useCallback(
+        async (newDescription: string) => {
+            if (description !== newDescription) {
+                await updateBlockSettings({ description: newDescription });
+            }
+        },
+        [description, updateBlockSettings],
+    );
+
+    const saveTitle = useCallback(
+        async (newTitle: string) => {
+            if (title !== newTitle) {
+                await updateBlockSettings({ title: newTitle });
+            }
+        },
+        [title, updateBlockSettings],
+    );
+
+    const onTemplateSelected = useCallback(
+        async (result: { template: TemplateLegacy }) => {
+            try {
+                await updateTemplateIdsFromKey(TEMPLATE_BLOCK_SETTING_ID, [result.template.id]);
+                updateBlockSettings({
+                    template: result.template,
+                    templateId: result.template.id,
+                });
+                await saveTitle(convertToRteValue(TextStyles.heading3, result.template.title));
+                await saveDescription(result.template.description);
+                setTemplateTextKey(generateRandomId());
+            } catch (error) {
+                setLastErrorMessage(error as string);
+            }
+
+            await appBridge.dispatch(closeTemplateChooser());
+        },
+        [appBridge, saveDescription, saveTitle, updateBlockSettings, updateTemplateIdsFromKey],
+    );
+
+    useEffect(() => {
+        const unsubscribeTemplateChooser = appBridge.subscribe('templateChosen', onTemplateSelected);
+
+        return () => {
+            if (typeof unsubscribeTemplateChooser === 'function') {
+                unsubscribeTemplateChooser();
+            }
+        };
+    }, [appBridge, onTemplateSelected]);
+
+    useEffect(() => {
+        if (error !== null) {
+            setLastErrorMessage(error);
+        }
+    }, [error]);
+
+    useEffect(() => {
+        if (blockTemplates[TEMPLATE_BLOCK_SETTING_ID] && blockTemplates[TEMPLATE_BLOCK_SETTING_ID].length > 0) {
+            const lastTemplate = blockTemplates[TEMPLATE_BLOCK_SETTING_ID].pop();
+
+            if (lastTemplate) {
+                setSelectedTemplate(lastTemplate);
+            }
+        }
+    }, [blockTemplates]);
+
+    return {
```

Comment:
Just a suggestion but I wonder if it makes sense with this many props to split these into relevant sub-objects. Also does the end user need all data such as "preview, previewCustom and hasPreview"? Should these properties be resolved into a single consumable value before returning?

Reply:
Yea I agree and considered this as well. I will try and improve how the data is organized in later branches